include/crow/routing.h

```diff
@@ -1340,7 +1363,7 @@ namespace crow
                 current_mw.merge_back(blueprint->mw_indices_);
                 for (auto& rule : blueprint->all_rules_)
                 {
-                    if (rule)
+                    if (rule && !rule->is_added())
```

Comment:
I think you're missing a similar check there (just above):
```c++
                if (blueprint->static_dir_ == "" && blueprint->all_rules_.empty())
                {
                    std::vector<HTTPMethod> methods;
                    get_recursive_child_methods(blueprint, methods);
                    for (HTTPMethod x : methods)
                    {
                        int i = static_cast<int>(x);
                        per_methods_[i].trie.add(blueprint->prefix(), 0, blueprint->prefix().length(), i);
                    }
                }
```

It seems to me that `per_methods_[i].trie.add(...)` will keep adding routes here, if `validate_bp()` is called multiple times.

Reply:
In `Blueprint` I implemented a flag like the one for `BaseRule`. It seems enough to me

Reply:
Take a look at my edits: in `Blueprint` I implemented a flag like the one for `BaseRule`. It seems enough to me